### DISCUSSION TITLE
Allow using strict on jinja2 overriden templates

### DISCUSSION THREAD

**Discussion by ddorian** [mkdocs/mkdocs#3623] [Q&A] [answered]
Hi,

I would like to enable strict jinja2 when trying to print `{{ config.extra.missing_variable }}` in overridden templates and get an error instead of a blank string. Is this currently possible? The `--strict` flag doesn't help.

**Comment by lovelydinosaur** [comment]
> Is this currently possible?

It isn't.

**Reply by ddorian** [reply]
Maybe it can be added as a feature request?

**Comment by kamilkrzyskow** [comment]
Would modifying the Environment.undefined class in the on_env event not allow to change the default behaviour?

There is also the option to override the get_template function on the Environment object as well, but this is more invasive.

This could be done with a Python hook. Never tried it though ✌️

**Comment by pawamoy** [comment, marked as answer]
~~Looks like the discussion is closed (doesn't appear in the timeline though, not great UX), but I'll discard the discussion state and comment further.~~ Found it, it's in "Events" in the right sidebar :slightly_smiling_face: 

IMO this would be a useful configuration option. However this can be implemented as a plugin so I don't think we should include this option in the core. A plugin can simply hook into [`on_env`](https://www.mkdocs.org/dev-guide/plugins/#on_env) to alter the Jinja environment and set its `undefined` attribute to `StrictUndefined` (see Jinja's docs).
